plugins/factory-plugin/src/theia-commands.ts

```diff
@@ -28,7 +29,33 @@ function isDevfileProjectConfig(project: cheApi.workspace.ProjectConfig | cheApi
     return !!project.name && !!project.source && !!project.source.type && !!project.source.location && !project.source['parameters'];
 }
 
-export class TheiaCloneCommand {
+export interface TheiaCloneCommand {
+    execute(): PromiseLike<void>;
+}
+
+export function getTheiaCloneCommand(
+    project: cheApi.workspace.ProjectConfig | cheApi.workspace.devfile.Project,
+    projectsRoot: string
+): TheiaCloneCommand {
+
+    if (!project.source) {
+        return;
+    }
+
+    switch (project.source.type) {
+        case 'git':
+            return new TheiaGitCloneCommand(project, projectsRoot);
+        case 'zip':
+            return new TheiaZipCloneCommand(project, projectsRoot);
+        default:
+            const message = `Project type "${project.source.type}" is not supported yet.`;
```

Comment:
Maybe without `yet`? With `yet` it sounds like we are promising to support any source type.

Reply:
Agree, will fix. Thanks